Game crashes when click the download button

Click this button in the mod menu, then the game will crash.
![image](https://user-images.githubusercontent.com/59791306/130541553-37afaf6b-a8c4-43d7-9578-590a0dc7f56d.png)

Error logs: 
[Crash Report-2021-8-24_9.31.00.zip](https://github.com/DeathsGun/ModManager/files/7035838/Crash.Report-2021-8-24_9.31.00.zip)

```
[21:39:02] [Render thread/INFO]: Getting categories
[21:39:03] [Render thread/FATAL]: Reported exception thrown!
net.minecraft.class_148: mouseClicked event handler
	at net.minecraft.class_437.method_25412(class_437.java:495) ~[intermediary-minecraft-1.17.1-client.jar:?]
	at net.minecraft.class_312.method_1601(class_312.java:94) ~[intermediary-minecraft-1.17.1-client.jar:?]
	at net.minecraft.class_312.method_22686(class_312.java:165) ~[intermediary-minecraft-1.17.1-client.jar:?]
	at net.minecraft.class_1255.execute(class_1255.java:101) ~[intermediary-minecraft-1.17.1-client.jar:?]
	at net.minecraft.class_312.method_22684(class_312.java:165) ~[intermediary-minecraft-1.17.1-client.jar:?]
	at org.lwjgl.glfw.GLFWMouseButtonCallbackI.callback(GLFWMouseButtonCallbackI.java:36) ~[lwjgl-glfw-3.2.2.jar:build 10]
	at org.lwjgl.system.JNI.invokeV(Native Method) ~[lwjgl-3.2.2.jar:build 10]
	at org.lwjgl.glfw.GLFW.glfwWaitEventsTimeout(GLFW.java:3174) ~[lwjgl-glfw-3.2.2.jar:build 10]
	at com.mojang.blaze3d.systems.RenderSystem.limitDisplayFPS(RenderSystem.java:190) ~[intermediary-minecraft-1.17.1-client.jar:?]
	at net.minecraft.class_310.method_1523(class_310.java:1140) ~[intermediary-minecraft-1.17.1-client.jar:?]
	at net.minecraft.class_310.method_1514(class_310.java:728) [intermediary-minecraft-1.17.1-client.jar:?]
	at net.minecraft.client.main.Main.main(Main.java:217) [intermediary-minecraft-1.17.1-client.jar:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:78) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:567) ~[?:?]
	at net.fabricmc.loader.game.MinecraftGameProvider.launch(MinecraftGameProvider.java:234) [fabric-loader-0.11.6.jar:?]
	at net.fabricmc.loader.launch.knot.Knot.launch(Knot.java:153) [fabric-loader-0.11.6.jar:?]
	at net.fabricmc.loader.launch.knot.KnotClient.main(KnotClient.java:28) [fabric-loader-0.11.6.jar:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:78) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:567) ~[?:?]
	at org.multimc.onesix.OneSixLauncher.launchWithMainClass(OneSixLauncher.java:210) [NewLaunch.jar:?]
	at org.multimc.onesix.OneSixLauncher.launch(OneSixLauncher.java:245) [NewLaunch.jar:?]
	at org.multimc.EntryPoint.listen(EntryPoint.java:143) [NewLaunch.jar:?]
	at org.multimc.EntryPoint.main(EntryPoint.java:34) [NewLaunch.jar:?]
Caused by: java.lang.AbstractMethodError: Receiver class xyz.deathsgun.modmanager.gui.ModsOverviewScreen does not define or inherit an implementation of the resolved method 'abstract net.minecraft.class_364 getFocused()' of interface xyz.deathsgun.modmanager.gui.widget.better.IListScreen.
	at xyz.deathsgun.modmanager.gui.widget.better.BetterListWidget.method_25316(BetterListWidget.java:52) ~[modmanager-1.0.0-alpha.jar:?]
	at net.minecraft.class_350.method_37018(class_350.java:541) ~[intermediary-minecraft-1.17.1-client.jar:?]
	at net.minecraft.class_437.method_37061(class_437.java:603) ~[intermediary-minecraft-1.17.1-client.jar:?]
	at net.minecraft.class_437.method_37056(class_437.java:581) ~[intermediary-minecraft-1.17.1-client.jar:?]
	at net.minecraft.class_437.method_37062(class_437.java:576) ~[intermediary-minecraft-1.17.1-client.jar:?]
	at net.minecraft.class_6385.method_37044(class_6385.java:34) ~[intermediary-minecraft-1.17.1-client.jar:?]
	at net.minecraft.class_437.method_37065(class_437.java:566) ~[intermediary-minecraft-1.17.1-client.jar:?]
	at net.minecraft.class_437.method_37064(class_437.java:561) ~[intermediary-minecraft-1.17.1-client.jar:?]
	at net.minecraft.class_437.method_25423(class_437.java:388) ~[intermediary-minecraft-1.17.1-client.jar:?]
	at net.minecraft.class_310.method_1507(class_310.java:984) ~[intermediary-minecraft-1.17.1-client.jar:?]
	at com.terraformersmc.modmenu.gui.ModsScreen.md30ffe7$lambda$onInit$0$0(ModsScreen.java:594) ~[modmenu-2.0.5.jar:?]
	at net.minecraft.class_4185.method_25306(class_4185.java:40) ~[intermediary-minecraft-1.17.1-client.jar:?]
	at net.minecraft.class_4264.method_25348(class_4264.java:16) ~[intermediary-minecraft-1.17.1-client.jar:?]
	at net.minecraft.class_339.method_25402(class_339.java:121) ~[intermediary-minecraft-1.17.1-client.jar:?]
	at net.minecraft.class_4069.method_25402(class_4069.java:27) ~[intermediary-minecraft-1.17.1-client.jar:?]
	at net.minecraft.class_312.method_1611(class_312.java:94) ~[intermediary-minecraft-1.17.1-client.jar:?]
	at net.minecraft.class_437.method_25412(class_437.java:490) ~[intermediary-minecraft-1.17.1-client.jar:?]
	... 26 more
---- Minecraft Crash Report ----
// Surprise! Haha. Well, this is awkward.
```

```
Getting categories
11:23:44.846
Reported exception thrown!
net.minecraft.class_148: mouseClicked event handler
	at net.minecraft.class_437.method_25412(class_437.java:495)
	at net.minecraft.class_312.method_1601(class_312.java:94)
	at net.minecraft.class_312.method_22686(class_312.java:165)
	at net.minecraft.class_1255.execute(class_1255.java:101)
	at net.minecraft.class_312.method_22684(class_312.java:165)
	at org.lwjgl.glfw.GLFWMouseButtonCallbackI.callback(GLFWMouseButtonCallbackI.java:36)
	at org.lwjgl.system.JNI.invokeV(Native Method)
	at org.lwjgl.glfw.GLFW.glfwPollEvents(GLFW.java:3101)
	at com.mojang.blaze3d.systems.RenderSystem.flipFrame(RenderSystem.java:165)
	at net.minecraft.class_1041.method_15998(class_1041.java:310)
	at net.minecraft.class_310.method_1523(class_310.java:1137)
	at net.minecraft.class_310.method_1514(class_310.java:728)
	at net.minecraft.client.main.Main.main(Main.java:217)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:78)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:567)
	at net.fabricmc.loader.game.MinecraftGameProvider.launch(MinecraftGameProvider.java:234)
	at net.fabricmc.loader.launch.knot.Knot.launch(Knot.java:153)
	at net.fabricmc.loader.launch.knot.KnotClient.main(KnotClient.java:28)
Caused by: java.lang.AbstractMethodError: Receiver class xyz.deathsgun.modmanager.gui.ModsOverviewScreen does not define or inherit an implementation of the resolved method 'abstract net.minecraft.class_364 getFocused()' of interface xyz.deathsgun.modmanager.gui.widget.better.IListScreen.
	at xyz.deathsgun.modmanager.gui.widget.better.BetterListWidget.method_25316(BetterListWidget.java:52)
	at net.minecraft.class_350.method_37018(class_350.java:541)
	at net.minecraft.class_437.method_37061(class_437.java:603)
	at net.minecraft.class_437.method_37056(class_437.java:581)
	at net.minecraft.class_437.method_37062(class_437.java:576)
	at net.minecraft.class_6385.method_37044(class_6385.java:34)
	at net.minecraft.class_437.method_37065(class_437.java:566)
	at net.minecraft.class_437.method_37064(class_437.java:561)
	at net.minecraft.class_437.method_25423(class_437.java:388)
	at net.minecraft.class_310.method_1507(class_310.java:984)
	at com.terraformersmc.modmenu.gui.ModsScreen.md6c367f$lambda$onInit$0$0(ModsScreen.java:594)
	at net.minecraft.class_4185.method_25306(class_4185.java:40)
	at net.minecraft.class_4264.method_25348(class_4264.java:16)
	at net.minecraft.class_339.method_25402(class_339.java:121)
	at net.minecraft.class_4069.method_25402(class_4069.java:27)
	at net.minecraft.class_312.method_1611(class_312.java:94)
	at net.minecraft.class_437.method_25412(class_437.java:490)
	... 19 more```

--- a/gradle.properties
+++ b/gradle.properties
@@ -22,7 +22,7 @@ minecraft_version=1.17.1
 yarn_mappings=1.17.1+build.39
 loader_version=0.11.6
 # Mod Properties
-mod_version=1.0.0-alpha
+mod_version=1.0.1-alpha
 maven_group=xyz.deathsgun
 archives_base_name=modmanager
 # Dependencies

--- a/src/main/java/xyz/deathsgun/modmanager/gui/ModsOverviewScreen.java
+++ b/src/main/java/xyz/deathsgun/modmanager/gui/ModsOverviewScreen.java
@@ -17,11 +17,13 @@
 package xyz.deathsgun.modmanager.gui;
 
 import net.minecraft.client.font.TextRenderer;
+import net.minecraft.client.gui.Element;
 import net.minecraft.client.gui.screen.Screen;
 import net.minecraft.client.gui.widget.ButtonWidget;
 import net.minecraft.client.gui.widget.TextFieldWidget;
 import net.minecraft.client.util.math.MatrixStack;
 import net.minecraft.text.TranslatableText;
+import org.jetbrains.annotations.Nullable;
 import xyz.deathsgun.modmanager.gui.widget.CategoryListEntry;
 import xyz.deathsgun.modmanager.gui.widget.CategoryListWidget;
 import xyz.deathsgun.modmanager.gui.widget.ModListEntry;
@@ -122,6 +124,12 @@ public class ModsOverviewScreen extends Screen implements IListScreen {
                 }
             }
         }
+    }
+
+    @Nullable
+    @Override
+    public Element getFocused() {
+        return super.getFocused();
     }
 
     @SuppressWarnings("unchecked")

--- a/src/main/java/xyz/deathsgun/modmanager/services/UpdateCheckService.java
+++ b/src/main/java/xyz/deathsgun/modmanager/services/UpdateCheckService.java
@@ -101,7 +101,7 @@ public class UpdateCheckService extends Thread {
                 latestVersion = version;
             }
         }
-        if (latest == null || installedVersion.compareTo(latestVersion) == 0) {
+        if (latest == null || installedVersion.compareTo(latestVersion) > 0) {
             logger.info("No update for {} found!", container.getId());
             return null;
         }

